docs/source/prjxray/vtr-rrgraph.rst

```diff
@@ -0,0 +1,404 @@
+   and to VPR channels?
+
+      `prjxray_assign_tile_pin_direction.py`_
+
+After the preparation work, output can be generated for VPR. 3 types of
+output are generated:
+
+-  Tile pb_types XML's that connect site pb_types **site pins** to
+   **tile wires**
+
+      `prjxray_tile_import.py`_
+
+-  Architecture XML that is the grid and has direct inter-tile
+   connections
+
+      `prjxray_arch_import.py`_
+
+-  Final routing graph XML
+
+      `prjxray_routing_import.py`_
+
+.. _prjxray_form_channels.py: https://github.com/SymbiFlow/symbiflow-arch-defs/blob/master/xc7/utils/prjxray_form_channels.py
+.. _prjxray_assign_tile_pin_direction.py: https://github.com/SymbiFlow/symbiflow-arch-defs/blob/master/xc7/utils/prjxray_assign_tile_pin_direction.py
+.. _prjxray_tile_import.py: https://github.com/SymbiFlow/symbiflow-arch-defs/blob/master/xc7/utils/prjxray_tile_import.py
+.. _prjxray_arch_import.py: https://github.com/SymbiFlow/symbiflow-arch-defs/blob/master/xc7/utils/prjxray_arch_import.py
+.. _prjxray_routing_import.py: https://github.com/SymbiFlow/symbiflow-arch-defs/blob/master/xc7/utils/prjxray_routing_import.py
+
+Click on the figure below to zoom-in:
+
+.. thumbnail:: ../images/prjxray/import-flow.png
+
+   This diagram shows the importing flow for Project X-Ray.
+
+Tile wire classification
+------------------------
+
+Before channels can be formed, **tile wires** need to be bucketed into
+their purpose.
+
+Step (1) - Group tile wires into “nodes”
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+   The first step is to first re-form nodes that contain all the
+   directly connected **tiles wires** These nodes are *not* VPR
+   **nodes**, they are simply the collection of **tile wires** that are
+   already a net (electrically equivalent).
+
+Step (2) - Classify “nodes”
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+   Each node then needs to be classified. The simplest classification is
+   a channel wire, which means that **pips** route on and off of the
+   node. However there are other important classifications. For example,
+   the carry chain connection between two CLBLL_L tiles should be
+   modelled as a tile direct connection, rather than routing onto a
+   channel and back off. The is classified as a "edge with mux".
+
+The classification is broken down into the following categories:
+
+-  CHANNEL - **Pips** route on and off of this node.
+
+-  EDGE_WITH_MUX - Two **tile wires** connected by a **pip**.
+
+   -  The first **tile wire** sources at a **site pin**, and the second
+      **tile wire** sinks at a **site pin**.
+
+   -  This captures direct inter-tile connections like carry chain
+      wires, BRAM data cascade wires, etc.
+
+-  NULL - A node that has either no source or no sink. This wires
+   typically occur near the edge of the grid.
+
+-  EDGES_TO_CHANNEL - A node that sources and sinks from a **site pin**
+   and connects via a **pip** to a CHANNEL
+
+.. figure:: ../images/prjxray/import-wire-class.svg
```

Comment:
This image not working?

Reply:
why?

![rtd-img](https://user-images.githubusercontent.com/52699314/76356978-20911c80-6317-11ea-8999-05a021e5d2f8.png)

Reply:
It appears to be working now